subprojects/model-core/src/main/java/org/gradle/internal/metaobject/BeanDynamicObject.java

```diff
@@ -108,15 +111,32 @@ public BeanDynamicObject(Object bean, @Nullable Class<?> publicType) {
         this.delegate = determineDelegate(bean);
     }
 
+    private static final MethodHandle ADD_INVOCATION_HOOKS_TO_META_CLASS_METHOD;
+
+    static {
+        try {
+            Class<?> metaClassHelperClass = Class.forName("org.gradle.internal.classpath.InstrumentedGroovyMetaClassHelper");
```

Comment:
❓ Are we using reflection here due to a cycle or?

Reply:
Yes, it's `core` → `model-core` but here we need to depend the other way.